pkg/verify/dsse_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package verify
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/secure-systems-lab/go-securesystemslib/dsse"
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/stretchr/testify/assert"
+)
+
+type dsseSigner struct {
+	pk *ecdsa.PrivateKey
+}
+
+func (d *dsseSigner) Sign(_ context.Context, data []byte) ([]byte, error) {
+	digest := sha256.Sum256(data)
+	return d.pk.Sign(rand.Reader, digest[:], nil)
+}
+
+func (d *dsseSigner) KeyID() (string, error) {
+	return "", nil
+}
+
+type envelopeContent struct {
+	EnvelopeContent
+	e *dsse.Envelope
+}
+
+func (e *envelopeContent) RawEnvelope() *dsse.Envelope {
+	return e.e
+}
+
+func TestVerifyEnvelope(t *testing.T) {
```

Comment:
I would like to be explicit what the purpose of the test is, for example (or add a comment if you prefer):
```suggestion
func TestVerifyEnvelopeSignatureCount(t *testing.T) {
```

Reply:
Yeah, good call out. Name is too generic.